packages/fetchai/contracts/erc1155/contract.py

```diff
@@ -18,20 +18,464 @@
+            terms.to_address,
+            terms.item_ids,
+            terms.from_supplies,
+            terms.to_supplies,
+            terms.value_eth_wei,
+            terms.trade_nonce,
+            signature,
+            data,
+        ).buildTransaction(
+            {
+                "chainId": 3,
+                "gas": 2818111,
+                "from": terms.from_address,
+                "value": terms.value_eth_wei,
+                "gasPrice": ledger_api.api.toWei("50", "gwei"),
+                "nonce": nonce,
+            }
+        )
+        return tx
+
+    def _get_balance(self, contract, from_address: Address, item_id: int):
+        """Get the balance for the specific id."""
+        return contract.instance.functions.balanceOf(
+            from_address, item_id
+        ).call()
+
+    def atomic_swap_single(self, contract, terms, signature) -> TransactionMessage:
+        """Make a trustless trade between to agents for a single token."""
+        assert self.address == terms.from_address, "Wrong from address"
+
+        tx = contract.get_trade_tx(terms=terms, signature=signature)
+
+        tx_message = TransactionMessage(
+            performative=TransactionMessage.Performative.PROPOSE_FOR_SIGNING,
+            skill_callback_ids=[ContractId("author", "my_skill", "0.1.0")],
+            tx_id="contract_deployment",
+            tx_sender_addr=terms.from_address,
+            tx_counterparty_addr="",
+            tx_amount_by_currency_id={"ETH": 0},
+            tx_sender_fee=0,
+            tx_counterparty_fee=0,
+            tx_quantities_by_good_id={},
+            info={},
+            ledger_id="ethereum",
+            signing_payload=tx
+        )
+
+        return tx_message
+
+    def _get_balance_of_batch(self, contract, address, item_ids):
+        """Get the balance for a batch of items"""
+        return contract.instance.functions.balanceOfBatch(
+            [address] * 10, item_ids
+        ).call()
+
+    def atomic_swap_batch(self, deployer_address, contract, terms, signature) -> TransactionMessage:
+        """Make a trust-less trade for a batch of items between 2 agents."""
+        assert deployer_address == terms.from_address, "Wrong 'from' address"
+        tx = contract.get_trade_batch_tx(terms=terms, signature=signature)
+
+        tx_message = TransactionMessage(
+            performative=TransactionMessage.Performative.PROPOSE_FOR_SIGNING,
+            skill_callback_ids=[ContractId("author", "my_skill", "0.1.0")],
+            tx_id="contract_deployment",
+            tx_sender_addr=terms.from_address,
+            tx_counterparty_addr="",
+            tx_amount_by_currency_id={"ETH": 0},
+            tx_sender_fee=0,
+            tx_counterparty_fee=0,
+            tx_quantities_by_good_id={},
+            info={},
+            ledger_id="ethereum",
+            signing_payload=tx
+        )
+
+        return tx_message
+
+    def sign_single_transaction(self, terms) -> TransactionMessage:
```

Comment:
it should be called `get_single_transaction_proposal`, or something like that

Reply:
It is more of the signing payload here. And not about the proposal. This is the second part that we sign the items we want to trade. So I wouldn't say that this is a transaction_proposal. More like `get_single_signing_proposal`. What do you think ?

Reply:
ok